syft/internal/unionreader/union_reader.go

```diff
@@ -44,6 +45,17 @@ func GetUnionReader(readerCloser io.ReadCloser) (UnionReader, error) {
 		return reader, nil
 	}
 
+	// file.LocationReadCloser embeds a ReadCloser, which is likely
+	// to implement UnionReader. Check whether the embedded read closer
+	// implements UnionReader, and just return that if so.
+	r, ok := readerCloser.(file.LocationReadCloser)
+	if ok {
+		ur, ok := r.ReadCloser.(UnionReader)
```

Comment:
I'm not 100% certain whether this also needs to check for `nil`. Would be happy for some reviewer input here.

Reply:
Since `file.LocationReadCloser` is a _struct_ and not a pointer (not `*file.LocationReadCloser`), it won't ever be `nil`, the check you have is right 👍 .

Reply:
But the io.ReadCloser on it could be nil, right?

Reply:
Ah yes, there's definitely a possibility it could be nil. You could probably _just_ check for that, e.g. `ur, _ := r.ReadCloser...`

Reply:
This behavior isn't different now - `GetUnionReader` has always been able to accept and return `nil`, so let's just leave this up to the caller.